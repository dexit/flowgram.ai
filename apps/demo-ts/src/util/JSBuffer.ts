export class JSBuffer {
  buffer: string[] = []

  constructor(public indent = '') {}

  clear(): JSBuffer {
    this.buffer = []
    return this
  }

  push(...args: string[]): JSBuffer {
    this.buffer.push(...args)
    return this
  }

  pushLine(...args: string[]): JSBuffer {
    this.buffer.push(...args, '\n')
    return this
  }

  toJs(separator: string = ''): string {
    // TODO js string perf
    // FIXME last line indent
    return (
      this.buffer
        .join(separator)
        .split('\n')
        .map((line) => `${this.indent}${line}`)
        // .map((line) => (line !== '\n' ? `${this.indent}${line}` : line))
        .join('\n')
    )
  }

  toString(): string {
    return this.toJs()
  }
}
