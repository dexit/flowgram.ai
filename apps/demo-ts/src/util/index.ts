export * from './log'
export * from './JSBuffer'
