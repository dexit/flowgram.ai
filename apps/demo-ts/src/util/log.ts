import debug from 'debug'

export const info = debug('log:info')
export const log = info
export const warn = debug('log:warn')
export const error = debug('log:error')
