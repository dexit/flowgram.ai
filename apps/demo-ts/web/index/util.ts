
export const codeMirrorBaseConfig: CodeMirror.EditorConfiguration = Object.freeze({
  mode: { name: 'javascript', json: true },
  lineNumbers: true,
  lineWrapping: false,
  theme: 'twilight',
  foldGutter: true,
  gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
  readOnly: true,
  cursorBlinkRate: -1,
})