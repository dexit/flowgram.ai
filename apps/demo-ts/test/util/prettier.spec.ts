import * as prettier from 'prettier'
import { prettierConfig } from '../flow/util'

describe('prettier', () => {
  it('basic', async () => {
    expect(await prettier.format('foo ( );', prettierConfig)).toMatchSnapshot()
    expect(
      await prettier.format('function foo (a , b=1 ){ f( a)}', prettierConfig)
    ).toMatchSnapshot()
  })

  it('module', async () => {
    expect(
      await prettier.format(
        `
export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    query: 'hi',
  }
  Object.assign(start.outputs, __args)
    //=== variable@variable | {level:1,blockIndex:1}
  const variable = {}
  variable.outputs = {
    a: '',
  }
  variable.outputs.a = start.outputs.query
    //=== end@end | {level:1,blockIndex:2}
  const end = {}
  end.data = {
    data: variable.outputs.a,
  }
  return end.data
  }
`,
        prettierConfig
      )
    ).toMatchSnapshot()
  })
})
