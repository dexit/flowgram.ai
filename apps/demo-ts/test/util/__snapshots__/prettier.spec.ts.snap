// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prettier basic 1`] = `
"foo()
"
`;

exports[`prettier basic 2`] = `
"function foo(a, b = 1) {
  f(a)
}
"
`;

exports[`prettier module 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    query: 'hi',
  }
  Object.assign(start.outputs, __args)
  //=== variable@variable | {level:1,blockIndex:1}
  const variable = {}
  variable.outputs = {
    a: '',
  }
  variable.outputs.a = start.outputs.query
  //=== end@end | {level:1,blockIndex:2}
  const end = {}
  end.data = {
    data: variable.outputs.a,
  }
  return end.data
}
"
`;
